.github/workflows/build.yml

```diff
@@ -1,10 +1,18 @@
-name: build
+name: Build
 
 on:
-  push:
-    branches:
-      - "main"
-  workflow_dispatch:
+  workflow_call:
+    inputs:
+      build-ref:
```

Comment:
This `ref` input is arguably the most important part of the PR 🙂 

> are triggered by a workflow_dispatch at the head of the selected branch (or push to main) so this would be fine.

This is exactly what we don't want. The are many people working on this repo at the same time, which means that a new commit can be pushed to a release branch at any moment, including during a release.

```
         Trigger
         release  Build
            |       |
main -*---*-+-*---*-+-*
      |     |     | |
      |     |  backport
      |     |     V |
      └-*---*-----*-+--
            ^       |
         Generate   |
          files     |
            |       |
```

In this scenario the build would happen on the backported commit, which may include, for example, changes to the UI that were not part of the generated files. So the final binary would have the wrong SHA **and** missing code.

By pinpointing to a specific commit SHA to build from, we can (more or less) guarantee that the final binary will be built from the expected commit that includes the generated files.

Another option would be what we did for the beta, which is to use a short-lived release branch, but I was told that this is not the recommended pattern for CRT.

We could also "freeze" branches during a release, preventing people from merging code, but this is confusing and cumbersome, so we would rather avoid it.

This was the only option left that I could think of.

Reply:
Ah apologies, I hadn't had my coffee yet and forgot that the generated files were committed 🙈 . Ignore my comment, this looks good. I think we would just need to make the update to pass the sha to the generate-release-metadata action. I can take a look at this today/tomorrow.